ledger/store/store.go

```diff
@@ -81,44 +79,67 @@ func NewStore(log zerolog.Logger, opts ...Option) (*Store, error) {
 		db:  db,
 		tx:  db.NewTransaction(true),
 
-		sema:  semaphore.NewWeighted(16),
-		err:   make(chan error, 16),
-		done:  make(chan struct{}),
-		mutex: &sync.RWMutex{},
-		wg:    &sync.WaitGroup{},
+		sema: semaphore.NewWeighted(16),
+		err:  make(chan error, 16),
+		done: make(chan struct{}),
+		txMu: &sync.RWMutex{},
+		wg:   &sync.WaitGroup{},
+
+		isDirtyMu: &sync.RWMutex{},
+		isDirty:   make(map[hash.Hash]bool, config.CacheSize),
 	}
 
 	s.wg.Add(1)
 	go s.flush()
 
-	s.cache, err = lru.NewWithEvict(config.CacheSize, func(k interface{}, v interface{}) {
+	s.cache, err = lru.NewWithEvict(config.CacheSize, func(k, v interface{}, used int) bool {
 		hash, ok := k.(hash.Hash)
 		if !ok {
 			logger.Fatal().Interface("got", k).Msg("unexpected key format")
 		}
 
-		value, ok := v.(ledger.Value)
-		if !ok {
-			logger.Fatal().Interface("got", v).Msg("unexpected value format")
+		// If the entry is dirty and the cache is not full, abort eviction.
+		s.isDirtyMu.RLock()
+		if s.isDirty[hash] && used < s.cacheSize {
+			s.isDirtyMu.RUnlock()
+			return false
```

Comment:
Isn't this redundant?

- if full and dirty, commit then evict
- otherwise, if dirty, simply don't evict
- otherwise, evict